### PR TITLE
Update tests with coverage docs with Gradle examples

### PR DESCRIPTION
**Pull request by brianwyka**
Reference: https://docs.gradle.org/7.3/userguide/jacoco_plugin.html

### CODE REVIEW

**Comment by brianwyka** [comment]
I think this needs some more work on my end.  Thought I had it working, but maybe not yet...  Still seeing some issues when running a combination of plain unit tests and tests with `@QuarkusTest` annotation

**Comment by brianwyka** [comment]
I'm struggling to get a combined report and the coverage verification to be aligned with the combination of plain unit tests and `@QuarkusTest`

**Comment by brianwyka** [comment]
Perhaps the `excludes` are not taken into account for the verification.  I think that may be the disconnect here.

File: docs/src/main/asciidoc/tests-with-coverage.adoc [hunk 1]
```diff
@@ -164,6 +164,7 @@ public class GreetingResourceTest {
 
 Now we need to add Jacoco to our project. To do this we need to add the following to the `pom.xml` dependencies section:
 
+.Maven
```
Comment (glefloch) [inline docs/src/main/asciidoc/tests-with-coverage.adoc:167]: I think this can be removed

File: docs/src/main/asciidoc/tests-with-coverage.adoc [hunk 2]
```diff
@@ -173,6 +174,12 @@ Now we need to add Jacoco to our project. To do this we need to add the followin
 </dependency>
 ----
 
+.Gradle
```
Comment (glefloch) [inline docs/src/main/asciidoc/tests-with-coverage.adoc:177]: I would more add a sentence like `If you are using Gradle, add the following dependency to our `build.gradle` file: `

File: docs/src/main/asciidoc/tests-with-coverage.adoc [hunk 3]
```diff
@@ -195,6 +202,7 @@ the coverage of other tests as well then you will need to fall back to the Jacoc
 
 In addition to including the `quarkus-jacoco` extension in your pom you will need the following config:
 
+.Maven
```
Comment (glefloch) [inline docs/src/main/asciidoc/tests-with-coverage.adoc:205]: I would also remove the title here

File: docs/src/main/asciidoc/tests-with-coverage.adoc [hunk 4]
```diff
@@ -239,6 +247,20 @@ In addition to including the `quarkus-jacoco` extension in your pom you will nee
 WARNING: This config will only work if at least one `@QuarkusTest` is being run. If you are not using `@QuarkusTest` then
 you can simply use the Jacoco plugin in the standard manner with no additional config.
 
+If you are using Gradle, add this to your `build.gradle`:
+
+.Gradle
```
Comment (glefloch) [inline docs/src/main/asciidoc/tests-with-coverage.adoc:252]: I would remove the title here. 
And in the snippet, we also need to add the jacoco plugin in the `plugins` block

File: docs/src/main/asciidoc/tests-with-coverage.adoc [hunk 5]
```diff
@@ -325,10 +347,13 @@ In order to run the integration tests as a jar with the Jacoco agent, add the fo
...
 WARNING: Sharing the same value for `quarkus.test.arg-line` might break integration test runs that test different types of Quarkus artifacts. In such cases, the use of maven profiles is advised.
 
-== Setting coverage thresholds for the Maven build
+== Setting coverage thresholds for the build
+
+=== Maven
 
 You can set thresholds for code coverage using the Jacoco Maven plugin. Note the element `<dataFile>${project.build.directory}/jacoco-quarkus.exec</dataFile>`. You must set it matching your choice for `quarkus.jacoco.data-file`.
 
+.Example Configuration
```
Comment (glefloch) [inline docs/src/main/asciidoc/tests-with-coverage.adoc:356]: I think I would remove the title here too

File: docs/src/main/asciidoc/tests-with-coverage.adoc [hunk 6]
```diff
@@ -375,6 +400,32 @@ You can set thresholds for code coverage using the Jacoco Maven plugin. Note the
 </build>
 ----
 
+=== Gradle
+Gradle has built-in `jacoco` plugin which allows you to update the `executionData` file within the `jacocoTestCoverageVerification` task.
+
+.Example Configuration
```
Comment (glefloch) [inline docs/src/main/asciidoc/tests-with-coverage.adoc:406]: Here I would also remove the title and add the application of the jacoco plugin in the snippet

**Review by glefloch — CHANGES_REQUESTED** [review]
Thanks for this @brianwyka. The overall looks good to me. I left some comments about titles and applying the `jacoco` plugin

**Comment by brianwyka** [comment]
Thanks for the review @glefloch .  I'll make the changes.  However, I think the section on the verification is still inaccurate since it does not work correctly at the moment.  Is there any idea there?

**Comment by brianwyka** [comment]
I believe I found a workaround to the issue (reported on Gradle):
https://github.com/gradle/gradle/issues/14760

Is it okay to document this workaround until its fixed on Gradle side?

**Comment by glefloch** [comment]
If it's working, I think we can with a reference to the issue

**Comment by brianwyka** [comment]
@glefloch , updated docs.

File: docs/src/main/asciidoc/tests-with-coverage.adoc [hunk 7]
```diff
@@ -375,6 +404,50 @@ You can set thresholds for code coverage using the Jacoco Maven plugin. Note the
...
+                value = 'COVEREDRATIO'
+                minimum = 0.72
+            }
+        }
+    }
+}
+check.dependsOn jacocoTestCoverageVerification
+----
+
+If you have any exclusions that you would like to apply to the verification task, that are outside the scope of `@QuarkusTest`, you will need the following in your `build.gradle` file:
```
Comment (glefloch) [inline docs/src/main/asciidoc/tests-with-coverage.adoc:432]:
```suggestion
Excluding classes from the verification task can be configured as following: 
```

File: docs/src/main/asciidoc/tests-with-coverage.adoc (hunk 6 above)
Comment (glefloch) [reply docs/src/main/asciidoc/tests-with-coverage.adoc:409]: Could you also remove these title ?

File: docs/src/main/asciidoc/tests-with-coverage.adoc (hunk 5 above)
Comment (glefloch) [reply docs/src/main/asciidoc/tests-with-coverage.adoc:360]: and this one too.

**Review by glefloch — COMMENTED** [review]
just left few comments. Should be good after that.

**Comment by brianwyka** [comment]
Should be all set now @glefloch

**Review by glefloch — APPROVED** [review]
Thanks @brianwyka

**Comment by gsmet** [comment]
I force pushed a rebase + squash. Will merge as soon as CI is green.

**Comment by glefloch** [comment]
Thanks @gsmet and @brianwyka !

**Pull request merged**